crates/orderbook/src/api/post_solver_competition.rs

```diff
@@ -0,0 +1,75 @@
+//! This is a private, undocumented api which will get replaced when we move the solution
+//! competition into the api.
+
+use crate::solver_competition::SolverCompetition;
+use anyhow::Result;
+use model::solver_competition::SolverCompetitionResponse;
+use reqwest::StatusCode;
+use std::{convert::Infallible, sync::Arc};
+use warp::{Filter, Rejection};
+
+fn request(
+) -> impl Filter<Extract = (String, u64, SolverCompetitionResponse), Error = Rejection> + Clone {
+    warp::post()
+        .and(warp::header::<String>("Authorization"))
+        .and(warp::path!("solver_competition" / u64))
+        .and(crate::api::extract_payload())
+}
+
+pub fn post(
+    handler: Arc<SolverCompetition>,
+    expected_auth: String,
+) -> impl Filter<Extract = (super::ApiReply,), Error = Rejection> + Clone {
+    let expected_auth = Arc::new(expected_auth);
+    request().and_then(
+        move |auth: String, auction_id: u64, model: SolverCompetitionResponse| {
+            let handler = handler.clone();
+            let expected_auth = expected_auth.clone();
+            async move {
+                let (json, status) = if auth.as_str() == expected_auth.as_str() {
+                    handler.set(auction_id, model);
+                    (warp::reply::json(&()), StatusCode::OK)
```

Comment:
Should this return a `No Content` or `Created` status code?

Reply:
Don't think it matters for our private temporary api so I didn't look into what the most proper  code is.

Reply:
Changed to Created